src/useq/_mda_sequence.py

```diff
@@ -323,21 +357,41 @@ def iter_sequence(sequence: MDASequence) -> Iterator[MDAEvent]:
     order = sequence.used_axes
 
     event_iterator = (enumerate(sequence.iter_axis(ax)) for ax in order)
+
     for global_index, item in enumerate(product(*event_iterator)):
         if not item:  # the case with no events
             continue
 
         _ev = dict(zip(order, item))
+
         index = {k: _ev[k][0] for k in INDICES if k in _ev}
 
         position: Optional[Position] = _ev[POSITION][1] if POSITION in _ev else None
         channel: Optional[Channel] = _ev[CHANNEL][1] if CHANNEL in _ev else None
         time: Optional[int] = _ev[TIME][1] if TIME in _ev else None
+        tile: Optional[dict[str, bool | float]] = _ev[TILE][1] if TILE in _ev else None
```

Comment:
`dict[str, bool | float]` isn't all that helpful from a typing perspective... (i.e. giving the Union of all possible value types isn't very helpful since you can't do much with it ... either just say `dict` and leave it as is or use a TypedDict